src/jsp/jsp_cl.cpp

```diff
@@ -1100,23 +1130,31 @@ drop_stored_procedure (const char *name, SP_TYPE_ENUM expected_type)
     }
 
   // delete _db_stored_procedure_code
-  err = db_get (sp_mop, SP_ATTR_TARGET, &target_val);
+
+  err = db_get (sp_mop, SP_ATTR_LANG, &lang_val);
   if (err != NO_ERROR)
     {
       goto error;
     }
 
-  target = db_get_string (&target_val);
-  // TODO: CBRD-24552
-  /*
-    class_name = get_class_name (target);
-
-  err = drop_stored_procedure_code (class_name.c_str ());
-  if (err != NO_ERROR)
+  lang = db_get_int (&lang_val);
+  if (lang == SP_LANG_PLCSQL)
     {
-      goto error;
+      err = db_get (sp_mop, SP_ATTR_TARGET, &target_val);
+      if (err != NO_ERROR)
+	{
+	  goto error;
+	}
+
+      target = db_get_string (&target_val);
+      class_name = get_class_name (std::string (target));
+
```

Comment:
no need to check null string?

Reply:
it is guaranteed that the target is not NULL.